Out of the box, `./build.sh` fails on arm64 with this error:

```
/home/omajid/.nuget/packages/microsoft.net.sdk.il/5.0.0-alpha1.19413.7/targets/Microsoft.NET.Sdk.IL.targets(135,5): error MSB3073: The command "/home/omajid/corefx/artifacts/toolset/ilasm/ilasm -QUIET -NOLOGO -DLL  -INCLUDE=include/netcoreapp -OUTPUT="/home/omajid/corefx/artifacts/obj/System.Runtime.CompilerServices.Unsafe/netcoreapp-Debug/System.Runtime.CompilerServices.Unsafe.dll" -KEY="/home/omajid/.nuget/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.19416.16/tools/snk/MSFT.snk" System.Runtime.CompilerServices.Unsafe.il" exited with code 126. [/home/omajid/corefx/src/System.Runtime.CompilerServices.Unsafe/src/System.Runtime.CompilerServices.Unsafe.ilproj]  
```

it turns out the `ilasm` executible is bad:

```
$ file /home/omajid/corefx/artifacts/toolset/ilasm/ilasm
/home/omajid/corefx/artifacts/toolset/ilasm/ilasm: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=6b99261d576d77b0cb852e8d30e328a3a12f1e80, not stripped
```

It's incorrectly using the ilasm binary for x64 instead of the binary from arm64. 

A hardcoded fix for that is:

```diff
-    <ToolRuntimeRID>$(_runtimeOS)-x64</ToolRuntimeRID>
+    <ToolRuntimeRID>$(_runtimeOS)-arm64</ToolRuntimeRID>
 ```

But that will only work on arm64. If I generalize this to:


```diff
-    <ToolRuntimeRID>$(_runtimeOS)-x64</ToolRuntimeRID>
+    <ToolRuntimeRID>$(_runtimeOS)-$(ArchGroup)</ToolRuntimeRID>
 ```

Then that (AFAIK) breaks cross compilation - because we want to use x64 tools when building arm64 on x64.

So we need to use architecture of the platform that is running the build - `$(HostArch.ToString.ToLowerInvariant())` - here.

My fix then goes (maybe too far?) and sets ArchGroup (which can take values of `x86`, `x64`, `arm64` and `arm`) explicitly based on `HostArch`.

@tmds suggested offline that maybe we could only set `ArchGroup` to `$(HostArch)` on non-Windows platforms? That might take care of the issues with VS architecture?

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -64,9 +64,9 @@
     <TargetGroup Condition="'$(TargetGroup)' == ''">netcoreapp</TargetGroup>
     <OSGroup Condition="'$(OSGroup)' == ''">$(DefaultOSGroup)</OSGroup>
     <ConfigurationGroup Condition="'$(ConfigurationGroup)' == ''">Debug</ConfigurationGroup>
-    <HostArch>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture)</HostArch>
-    <ArchGroup Condition="'$(ArchGroup)' == '' AND '$(HostArch)' == 'Arm'">arm</ArchGroup>
-    <ArchGroup Condition="'$(ArchGroup)' == '' AND '$(HostArch)' == 'Arm64'">arm64</ArchGroup>
+    <HostArch>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</HostArch>
+    <ArchGroup Condition="'$(ArchGroup)' == '' AND '$(HostArch)' == 'arm'">arm</ArchGroup>
+    <ArchGroup Condition="'$(ArchGroup)' == '' AND '$(HostArch)' == 'arm64'">arm64</ArchGroup>
     <ArchGroup Condition="'$(ArchGroup)' == ''">x64</ArchGroup>
 
     <!-- Initialize BuildConfiguration from the individual properties if it wasn't already explicitly set -->
@@ -135,9 +135,10 @@
     <_runtimeOS Condition="'$(_runtimeOS)' == 'tizen.4.0.0'">linux</_runtimeOS>
     <_runtimeOS Condition="'$(_runtimeOS)' == 'tizen.5.0.0'">linux</_runtimeOS>
     <_runtimeOS Condition="'$(PortableBuild)' == 'true'">$(_portableOS)</_runtimeOS>
-    <ToolRuntimeRID>$(_runtimeOS)-x64</ToolRuntimeRID>
+    <ToolRuntimeRID Condition="'$(BuildingInsideVisualStudio)' == 'true'">$(_runtimeOS)-x64</ToolRuntimeRID>
+    <ToolRuntimeRID Condition="'$(ToolRuntimeID)' == ''">$(_runtimeOS)-$(HostArch)</ToolRuntimeRID>
     <!-- We build linux-musl-arm on a ubuntu container, so we can't use the toolset build for alpine runtime. We need to use portable linux RID for our toolset in order to be able to use it. -->
-    <ToolRuntimeRID Condition="'$(_runtimeOS)' == 'linux-musl' AND $(ArchGroup.StartsWith('arm')) AND !$(HostArch.StartsWith('Arm'))">linux-x64</ToolRuntimeRID>
+    <ToolRuntimeRID Condition="'$(_runtimeOS)' == 'linux-musl' AND $(ArchGroup.StartsWith('arm')) AND !$(HostArch.StartsWith('arm'))">linux-x64</ToolRuntimeRID>
 
     <!-- There are no WebAssembly tools, so treat them as Windows -->
     <ToolRuntimeRID Condition="'$(RuntimeOS)' == 'WebAssembly'">win-x64</ToolRuntimeRID>

